### PR TITLE
Player1 assigned on create, Player 2 on update, 1 error in Rspec test

### PR DESCRIPTION
**Pull request by waders92**
Able to create a game and have player_1_id assigned on CREATE as well as player_2_id assigned on UPDATE.
In writing the tests, the test for player 2 gives me a routing error.  Routes look good but will try to fix.  Celia wanted me to push this so she could take a look as well.

screen shot 2017-02-21 at 1 50 40 pm

### CODE REVIEW

File: spec/controllers/games_controller_spec.rb [hunk 1]
```diff
@@ -24,9 +24,37 @@
...
+      player_1 = FactoryGirl.create(:player)
+      sign_in player_1
+
+      post :create, params: { game: { name: 'test game' } }
+      @game = Game.last
+      sign_out player_1
+      player_2 = FactoryGirl.create(:player)
+      sign_in player_2
+
+      patch :update, id: @game.id, params: { game: { name: 'test game' } }
```
Comment (SijiaDavis) [inline spec/controllers/games_controller_spec.rb]: remove `params: ...... `

File: spec/controllers/games_controller_spec.rb [hunk 2]
```diff
@@ -24,9 +24,37 @@
...
+      @game.reload
+      expect(@game.name).to eq 'test game'
+      expect(@game.player_2_id).to eq player.id
```
Comment (SijiaDavis) [inline spec/controllers/games_controller_spec.rb]: change `player.id` to `player_2.id`

**Comment by waders92** [comment]
Don't delete this branch.  Matt and I are working on the Moving Pieces Card on this branch as well.  Thanks.

**Comment by waders92** [comment]
Latest push is for Matt to pull

**Comment by waders92** [comment]
4 Rubocop errors in code that had already been pushed.  I ran bundle exec rubocop and rubocop -a.  That left 4 issues.

File: app/views/pieces/show.html.erb [hunk 3]
```diff
@@ -14,4 +14,4 @@
   <% end %>
 </table>
 
-<% @piece.move_to(3,4) %>
+<!-- <% @piece.move_to(3,4) %> -->
```
Comment (mscoutermarsh) [inline app/views/pieces/show.html.erb]: can delete this line

**Comment by mscoutermarsh** [comment]
Take a look at the logs here to see whats wrong:

https://travis-ci.org/Coredump-FHP/chess/builds/206308276

File: app/helpers/pieces_helper.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module PiecesHelper
+  def render_piece_highlight(x, y)
+    piece = @game.retrieve_piece(x, y)
+    if !piece.nil?
```
Comment (waders92) [inline app/helpers/pieces_helper.rb]: I didn't write this code, I know what you are saying about "if/not" it's confusing - but I don't know where you are getting -
`return unless piece.present?`
I don't see that in the file.

Comment (mscoutermarsh) [inline app/helpers/pieces_helper.rb]: I mean, you can change it to `return unless piece.present?`

**Comment by waders92** [comment]
Mike this branch has the one rubocop error that myself and Celia can't seem to shake.  Ok to merge?  Tests are good to go.

**Comment by mscoutermarsh** [comment]
Check the logs

```
/home/travis/.rvm/gems/ruby-2.3.1/gems/activesupport-5.0.1/lib/active_support/dependencies.rb:263:in `rescue in load_dependency': /home/travis/build/Coredump-FHP/chess/app/helpers/pieces_helper.rb:13: syntax error, unexpected keyword_end, expecting end-of-input (SyntaxError)
```

That file has invalid Ruby. It's not just a rubocop error. It won't run at all.

File: app/helpers/pieces_helper.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  def render_piece_highlight(x, y)
+    piece = @game.retrieve_piece(x, y)
+    return unless piece.present?
+    if piece == @piece
+      return content_tag(:div, image_tag(piece.icon), class: 'highlight')
+    else
+      return image_tag(piece.icon)
+    end
+    link_to(raw(content_tag(:div, ' ', class: 'clickable-square')), piece_path(@piece, x_coordinate: x, y_coordinate: y), method: 'PUT')
+    end
```
Comment (mscoutermarsh) [inline app/helpers/pieces_helper.rb]: Delete this line

Comment (waders92) [inline app/helpers/pieces_helper.rb]: After I took out line 9, it told me that line 10 had unreachable code.  I removed it, not sure if I was supposed to.  I can put it back in if needed.

**Pull request merged**